frontend/src/pages/Root.js

```diff
@@ -0,0 +1,38 @@
+import { useEffect } from 'react';
+import { Outlet, useLoaderData, useSubmit } from 'react-router-dom';
+
+import MainNavigation from '../components/Header/MainNavigation';
+import { getTokenDuration } from '../util/auth';
+
+function RootLayout() {
+  const token = useLoaderData();
+  const submit = useSubmit();
+  useEffect(() => {
+    if (!token) {
+      return;
+    }
+
+    if (token === 'EXPIRED') {
```

Comment:
always constant for status checks like EXPIRED can be in a different file called constants like:
const EXPIRED_TOKEN = 'EXPIRED';